app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheet.kt

```diff
@@ -72,29 +79,44 @@ class BloodPressureEntrySheet : BottomSheetActivity(), BloodPressureEntryUi {
 
     fun intentForNewBp(context: Context, patientUuid: UUID): Intent {
       return Intent(context, BloodPressureEntrySheet::class.java)
-          .putExtra(KEY_OPEN_AS, OpenAs.New(patientUuid))
+          .putExtra(KEY_OPEN_AS, New(patientUuid))
     }
 
     fun intentForUpdateBp(context: Context, bloodPressureMeasurementUuid: UUID): Intent {
       return Intent(context, BloodPressureEntrySheet::class.java)
-          .putExtra(KEY_OPEN_AS, OpenAs.Update(bloodPressureMeasurementUuid))
+          .putExtra(KEY_OPEN_AS, Update(bloodPressureMeasurementUuid))
     }
 
     fun wasBloodPressureSaved(data: Intent): Boolean {
       return data.getBooleanExtra(EXTRA_WAS_BP_SAVED, false)
     }
   }
 
-  @SuppressLint("CheckResult")
-  override fun onCreate(savedInstanceState: Bundle?) {
-    super.onCreate(savedInstanceState)
+  @Inject
+  lateinit var crashReporter: CrashReporter
 
-    setContentView(R.layout.sheet_blood_pressure_entry)
-    TheActivity.component.inject(this)
+  private val viewRenderer = BloodPressureEntryViewRenderer(this)
 
-    bindUiToControllerWithoutDelay(
-        ui = this,
-        events = Observable.mergeArray(
+  private val delegate by unsafeLazy {
+    val defaultModel = when (val openAs = intent.extras!!.getParcelable(KEY_OPEN_AS) as OpenAs) {
+      is New -> BloodPressureEntryModel.newBloodPressureEntry(openAs.patientUuid)
```

Comment:
I don't understand the point of doing this since `BloodPressureEntryModel` just internally creates a new instance of the same `OpenAs` type anyway? We might as well just pass the `openAs` directly to the constructor of `BloodPressureEntryModel`.

Reply:
Makes sense, will fix this.